src/k8s/pkg/k8sd/app/app.go

```diff
@@ -242,3 +243,35 @@ func (a *App) markNodeReady(ctx context.Context, s state.State) {
 		}
 	}
 }
+
+func (a *App) waitKubernetesReady(ctx context.Context) error {
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		a.readyWg.Wait()
```

Comment:
we already have a `a.readyWg`, but now we add another layer on top of that. in that case, let's drop the waitgroup altogether and handle everything in one place.